### DISCUSSION TITLE
How to make sure AG.tabular uses GPU

### DISCUSSION THREAD

**Discussion by praveentitusf** [autogluon/autogluon#4754] [Q&A]
How can I make sure AG.tabular used GPU while training.

**Comment by LennartPurucker** [comment]
Heyho!

You can explicitly tell AutoGluon that a GPU is available by setting `num_gpus` during the `.fit()` call. However, AutoGluon does not use a GPU to train every model. Some models do not support GPU training (so far); for others, it would often be less efficient to use the available GPU(s) instead of the CPUs. 

Also, note that `presets='experimental_quality'` currently disables GPU usage. 

If you want to force to use a GPU for training every fold of every model, you can set the following in the `fit()` call:

```python
predictor = TabularPredictor(...).fit(
    ...
    ag_args_fit={
        "num_cpus": 0,
        "num_gpus": 1,
    },
)
```

Alternatively, you can set the ag_args_fit per model in the hyperparameters; see https://auto.gluon.ai/stable/api/autogluon.tabular.TabularPredictor.fit.html for more details.
